### PR TITLE
replace eventlet with gevent

### PR DESCRIPTION
**Pull request by chadrik**
I would love to be able to use `wampy` within my `gevent`-based application, but I'm pretty sure the use of `eventlet` precludes that.  I've never used `eventlet` so I can't speak to the technical differences between the two libraries, but `gevent` is *very* widely adopted at this point and `eventlet` is not.  Would you accept a PR that ports this project to `gevent`?   All the tests are passing.

### CODE REVIEW

**Comment by noisyboiler** [comment]
hi @chadrik 
i think the only reason i chose eventlet was because of familiarity with it having previously worked with nameko. It does look like gevent has the bigger community so it might make sense. I'd be interested to see your PR for sure and give it some thought.
Thanks for getting in touch.
Simon

**Comment by chadrik** [comment]
Glad to hear you're open to the idea.   As far as I can tell, the async-wars are now over and the two victors are asyncio (which killed twisted) and gevent (which killed eventlet), and given their completely different -- and so far incompatible -- approaches, there seems to be room enough for two at the top.

All the changes are here in this PR.  gevent is mostly a drop-in replacement for eventlet (after all gevent started as a [set of patches for eventlet](https://blog.gevent.org/2010/02/27/why-gevent/)).  One minor differences is that [`gevent.queue.Queue.get`](http://www.gevent.org/gevent.queue.html#gevent.queue.Queue.get) automatically wakes up waiting greenlets, so there's no need to loop and sleep (more concise and efficient that way).  Another is that it's good practice to give [`gevent.sleep`](http://www.gevent.org/gevent.html#gevent.sleep) a non-zero value to avoid an edge case where the same greenlet can be repeatedly rescheduled.

Another big potential advantage of adopting gevent is its wsgi support, for creating high performance WAMP clients behind a proxy server like nginx.  You can see an example of how to use `gevent.pywsgi` in [`gevent-websocket`](https://gitlab.com/noppo/gevent-websocket/), which AFAIK is the only websocket server that plays nice with gevent.  That project aims to be a general purpose websocket server, with WAMP support added as an optional feature, but I get the sense that its WAMP features are under-used and under-tested. It does test using the [`autobahn-testsuite`](https://github.com/crossbario/autobahn-testsuite), which I think would be a nice thing to add here but I can't yet tell if the project is testing its WAMP capabilities in particular (or if that's even possible with the testsuite).

Anyway, I'm impressed with your library.  I like code and the set of tests against crossbar (I may even use your pytest plugin in my app).  If I end up adopting the library I may also want to work with you on adding some other features, like WAMP v2 support.

Thanks!

File: wampy/session.py [hunk 1]
```diff
@@ -114,8 +115,8 @@ def send_message(self, message_obj):
 
     def recv_message(self, timeout=5):
         try:
-            message = self._wait_for_message(timeout)
-        except eventlet.Timeout:
+            message = self._message_queue.get(timeout=timeout)
+        except gevent.queue.Empty:
```
Comment (noisyboiler) [inline wampy/session.py]: 👍

**Comment by noisyboiler** [comment]
i'm sold. a big THANK YOU for this and for the time spent on this PR.
please add yourself to CONTRIBUTORS.txt and I'll do a release over the coming days.
Excited about working with someone on this too, so please stay in contact!
Simon

**Comment by noisyboiler** [comment]
👍

**Comment by keiser1080** [comment]
@chadrik please could you join the effort of @noisyboiler to add gevent support
refer to #https://github.com/noisyboiler/wampy/issues/71

**Pull request merged**